app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -214,6 +214,13 @@ def save_hosts_inventory(ems, hashes, target = nil)
     end
   end
 
+  def save_hosts_storages_inventory(host, hashes, target = nil)
+    hashes.each do |mount|
+      storage_id = mount[:storage][:id]
+      HostsStorage.where(:host_id => host.id, :storage_id => storage_id).update_all(:read_only => mount[:read_only])
```

Comment:
I ended up with update_all because other methods (save_inventory_multi and update_attributes both) failed to save to the DB due to lack of a primary_key on hosts_storages.

This was described as the way to update a join table that doesn't have a primary_key but for sure there are other solutions.  Would it be better to add an id column to the migration for hosts_storages?

For the record here is the error when using save_inventory_multi:
```
[----] I, [2015-09-25T15:25:26.478408 #30564:e3f990]  INFO -- : MIQ(ManageIQ::Providers::Vmware::InfraManager::HostEsx#raise_cluster_event) Raised EVM   Event: [host_add_to_cluster, host: per410b-t4.manageiq.com(1), cluster: Test Cluster(2)]
[----] E, [2015-09-25T15:25:27.371268 #30564:e3f990] ERROR -- : MIQ(EmsRefresh.save_hosts_inventory) EMS: [Test-VC50], id: [1] Processing Host: [per410b-t4.manageiq.com] failed with error [TypeError: nil is not a symbol nor a string]. Skipping Host. 
[----] E, [2015-09-25T15:25:27.371398 #30564:e3f990] ERROR -- : [TypeError]: nil is not a symbol nor a string  Method:[rescue in block in                save_hosts_inventory]
[----] E, [2015-09-25T15:25:27.371466 #30564:e3f990] ERROR -- : /home/remote/agrare/.gem/ruby/gems/activemodel-4.2.4/lib/active_model/dirty.rb:181:in    `attribute_was'
/home/remote/agrare/.gem/ruby/gems/activerecord-4.2.4/lib/active_record/attribute_methods/primary_key.rb:45:in `id_was'
/home/remote/agrare/.gem/ruby/gems/activerecord-4.2.4/lib/active_record/persistence.rb:515:in `_update_record'
...
```